### ISSUE TITLE
Installing extension from VSIX does not install extension

### ISSUE THREAD

**Issue by sguillia** [microsoft/vscode#83227]
Issue Type: **Bug**

**Steps to Reproduce - simplified**

1. Create and install a simple extension
2. Uninstall it
3. Change your extension source
4. Install it again
5. Observe as it didn't work

**Steps to Reproduce - detailed**

1. git clone https://github.com/microsoft/vscode-extension-samples
2. cp -R vscode-extension-samples/helloworld-minimal-sample ./
3. cd helloworld-minimal-sample
4. In package.json, change publisher to your name
5. vsce package
6. code-insiders --install-extension helloworld-minimal-sample-0.0.1.vsix
7. code-insiders --uninstall-extension helloworld-minimal-sample-0.0.1.vsix
8. In extension.js, replace `showInformationMessage('Hello World!')` by `showInformationMessage('Hello modified')`
9. vsce package
10. code-insiders --install-extension helloworld-minimal-sample-0.0.1.vsix
11. `code-insiders .`, run command `> Hello World`

**Expected**

The information message shows `Hello modified`

**What actually happens**

The information message shows `Hello World!`

**Work-around 1**

Install it another time fixes the problem
```
code-insiders --install-extension helloworld-minimal-sample-0.0.1.vsix
```

**Work-around 2**

Do not uninstall the extension, just run `--install-extension` and it works

**Work-around 3**

```
rm -rf ~/.vscode-insiders/extensions/sguillia.helloworld-minimal-sample-0.0.1
```
Note that `--uninstall` do not remove that folder.

**Work-around 4**

Change the version in package.json
This works, but it is usual for an extension author to regularly change an extension without updating the version. Also this one was hard to find out.

**Additional information**

- This is not a duplicate of #68234 since I open up a new vscode instance to test if the extension was installed.

- The extension is not available in the marketplace.

- When I run `code-insiders --install-extension` and it doesn't work, the command terminates early. It takes longer to install (half a second more) when it is actually installing the extension.

- Installing and uninstalling with `--verbose` doesn't print any additional message

Thank you

VS Code version: Code - Insiders 1.40.0-insider (f5d3ffa6a0d655c87e1eb0e1e90773df58f7ff25, 2019-10-23T05:26:16.965Z)
OS version: Linux x64 4.15.0-65-generic

**Comment by gjsjohnmurray** [comment]
> **Expected**
> 
> The information message shows `Hello World!`

I think you expected `Hello modified`

Anyway, in your detailed steps, if you close VSCode after step 7 and then launch it again does the folder disappear? It's been my experience that the cleanup of redundant extension folders happens only on the next startup.

**Comment by sguillia** [comment]
Indeed thanks

I run the commands from a terminal, so no vscode instance is open at that time.

When I open a vcode instance right after the uninstall, my extension folder doesn't disappear

**Comment by sandy081** [comment]
Installing from VSIX shall always replace the existing extension version.
